certbot-dns-cloudflare/certbot_dns_cloudflare/__init__.py

```diff
@@ -1 +1,76 @@
-"""Cloudflare DNS Authenticator"""
+"""
+The `~certbot_dns_cloudflare.dns_cloudflare` plugin automates the process of
+completing a ``dns-01`` challenge (`~acme.challenges.DNS01`) using the
+Cloudflare API.
+
+
+Named Arguments
+---------------
+
+========================================  =====================================
+``--dns-cloudflare-credentials``          Cloudflare credentials_ INI file.
+                                          (Required)
+``--dns-cloudflare-propagation-seconds``  The number of seconds to wait for DNS
+                                          to propagate before asking the ACME
+                                          server to verify the DNS record.
+                                          (Default: 10)
+========================================  =====================================
+
+Credentials
+-----------
+
+Use of this plugin requires a configuration file containing Cloudflare API
+credentials, obtained from your Cloudflare
+`account page <https://www.cloudflare.com/a/account/my-account>`_.
+
+.. code-block:: ini
+   :name: credentials.ini
+   :caption: Example credentials file:
+
+   # Cloudflare API credentials used by Certbot
+   dns_cloudflare_email = cloudflare@example.com
+   dns_cloudflare_api_key = 0123456789abcdef0123456789abcdef01234567
+
+The path to this file can be provided interactively or using the
+``--dns-cloudflare-credentials`` command-line argument. Certbot records the path
+to this file for use during renewal, but does not store the file's contents.
+
+.. caution::
+   You should protect these API credentials as you would the password to your
+   Cloudflare account. Users who can read this file can use these credentials
+   to issue API calls on your behalf. Users who can cause Certbot to run using
+   these credentials can complete a ``dns-01`` challenge to acquire new
+   certificates or revoke existing certificates for associated domains.
+
```

Comment:
We [warn](https://github.com/certbot/certbot/blob/master/certbot/plugins/dns_common.py#L304) if the file seems to have unsafe permissions, but beyond that I'm not sure what we should recommend at the moment.

One option would be to run certbot on a dedicated machine (with appropriate firewall rules, tightly controlled access, strict monitoring, etc.) and push certificates + private keys to the machines using the certificates via a post-renew hook, but that doesn't seem very elegant.

Once we can re-use private keys (#4610), the story gets a bit better, since private keys would no longer need to be sent over the wire following renewal.